relayer/app/cursor/cursor.go

```diff
@@ -0,0 +1,47 @@
+package cursor
+
+import (
+	"context"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/xchain"
+
+	ormv1alpha1 "cosmossdk.io/api/cosmos/orm/v1alpha1"
+	"cosmossdk.io/core/store"
+	"cosmossdk.io/orm/model/ormdb"
+	db "github.com/cosmos/cosmos-db"
+)
+
+func (c *Cursor) StreamID() xchain.StreamID {
+	return xchain.StreamID{
+		SourceChainID: c.GetSrcChainId(),
+		DestChainID:   c.GetDstChainId(),
+		ShardID:       xchain.ShardID(c.GetConfLevel()),
```

Comment:
used to fetch the cursors from network, want to confirm if it's ok for that usecase

Reply:
when fetching cursors from on-chain, we need to fetch all streams for the chain version, then use minimum attestOffset. We can't simply query the the first shard, there might be more and they might be lagging.